### PR TITLE
No need for raw data to run a project in headless

### PR DESCRIPTION
**Pull request by wolny**
Motivation:
unfortunately in order to run workflows in headless mode the raw data that the models were trained on has to be available. this has been quite uncomfortable for the users especially when the raw data has significant size (GBs) and has to be carry around in order to run tasks in headless.

Current solution:
- if we're in headless and the raw data exists the flow is the same as before
- if we're in headless and the raw data does not exists, tell the `opDataSelection` to fake the raw data and we are able to proceed with the batch
- if we're in headless, raw data does not exists and `--retrain` was passed -> we inform the user that retraining is not possible

Faking the raw data is possible, due to additional attributes to the lane metadata (`DatasetInfo`) serialized to the project file. The serialization is backward compatible, however because the 'old' project files lack the necessary metadata headless mode without raw data won't work (it obviously still work if the raw data is present, see 1st point above).

TODO:
- [x] Ensure backward compatibility
- [x] Extensive manual tests (different workflows, different dimensionality of raw data, multiple image lanes)
- [x] Write unit tests 
- [x] Write integration testes

Original issue: https://github.com/ilastik/ilastik/issues/1441

### CODE REVIEW

File: ilastik/applets/dataSelection/opDataSelection.py [hunk 1]
```diff
@@ -203,19 +209,6 @@ def filePath(self, newPath):
     def datasetId(self):
         return self._datasetId
 
-    DatasetInfoSchema = \
```
Comment (wolny) [inline ilastik/applets/dataSelection/opDataSelection.py]: It doesn't seems to be used anywhere, so I just deleted it as obsolete

File: ilastik/applets/dataSelection/opDataSelection.py [hunk 2]
```diff
@@ -343,13 +348,23 @@ def setupOutputs(self):
...
+                        opReader.SubVolumeRoi.setValue(datasetInfo.subvolume_roi)
+                    opReader.WorkingDirectory.setValue(self.WorkingDirectory.value)
+                    opReader.SequenceAxis.setValue(datasetInfo.sequenceAxis)
+                    opReader.FilePath.setValue(datasetInfo.filePath)
+                else:
+                    # Use fake reader: allows to run the project in a headless
+                    # mode without the raw data
+                    opReader = OpZeroDefault(parent=self)
+                    opReader.MetaInput.meta = MetaDict(shape=datasetInfo.laneShape,
+                                             dtype=datasetInfo.laneDtype,
```
Comment (FynnBe) [inline ilastik/applets/dataSelection/opDataSelection.py]: 👎 'manual linting bot alert'

File: ilastik/applets/dataSelection/dataSelectionSerializer.py [hunk 3]
```diff
@@ -493,6 +521,16 @@ def unload(self):
             This way we can avoid invalid state due to a partially loaded project. """ 
         self.topLevelOperator.DatasetGroup.resize( 0 )
 
+    def _shouldRetrain(self):
```
Comment (FynnBe) [inline ilastik/applets/dataSelection/dataSelectionSerializer.py]: maybe add a beautiful '@property' here?

File: ilastik/applets/dataSelection/opDataSelection.py (hunk 2 above)
Comment (wolny) [reply ilastik/applets/dataSelection/opDataSelection.py]: :crying_cat_face: :broken_heart:

Comment (wolny) [reply ilastik/applets/dataSelection/opDataSelection.py]: done :white_check_mark:

File: ilastik/applets/dataSelection/dataSelectionSerializer.py (hunk 3 above)
Comment (wolny) [reply ilastik/applets/dataSelection/dataSelectionSerializer.py]: this method should not be a part of  `dataSelectionSerializer` public API, so I'd rather keep it protected

Comment (FynnBe) [reply ilastik/applets/dataSelection/dataSelectionSerializer.py]: I don't see how '@property' would change anything about that. (It's always public anyway, the underscore can stay to indicate that it's meant to be private.)
Anyhow, it's fine as is 👍

Comment (wolny) [reply ilastik/applets/dataSelection/dataSelectionSerializer.py]: true! on the 2nd thought, I think the `@property` would be nice in here :)

Comment (wolny) [reply ilastik/applets/dataSelection/dataSelectionSerializer.py]: done :white_check_mark:

**Comment by wolny** [comment]
Unit and integration tests are in place now, so this is good to go from my side. Would be great to have it in in the next release, cause users have been waiting for this feature for a long time.

**Comment by wolny** [comment]
Thanks for the review @FynnBe!

**Pull request merged**